compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2009,7 +2017,14 @@ object Parsers {
           val uscoreStart = in.skipToken()
           if (isIdent(nme.raw.STAR)) {
             in.nextToken()
-            if (in.token != RPAREN) syntaxError(SeqWildcardPatternPos(), uscoreStart)
+            if in.token != RPAREN || !inArguments && sourceVersion.isAtLeast(`3.1`) then
+              syntaxError(SeqWildcardPatternPos(), uscoreStart)
```

Comment:
Maybe be more aggressive and demand this for 3.0? In any case, we should do migration warning and it would be nice to also offer a rewrite.

Reply:
In fact, I believe the warning that this is no longer supported is more important than the error that it's in the wrong place. The theory is that any code that has `xs @ _*` will have already been compiled by Scala 2 (in most cases, at least)